### PR TITLE
Java: do not generate models for `toString` and lambda flow methods

### PR DESCRIPTION
**Pull request by owen-mc**
We generally don't want to model `toString`. It is better not to generate models for them, and add manual models where they are needed.

We also don't want to have any models for certain functions relating to lambda flow.

### CODE REVIEW

**Comment by michaelnebel** [comment]
@owen-mc : Looks like you pushed the wrong branch :-D

**Comment by owen-mc** [comment]
Oops. This is what happens when I work late at night.

**Review by atorralba — DISMISSED** [review]

**Review by michaelnebel — DISMISSED** [review]
👍

**Comment by owen-mc** [comment]
I've added an exclusion for the functions in the comment at the bottom of `java/ql/src/Metrics/Summaries/TopJdkApis.qll` which relate to lambda flow. I will remove them from https://github.com/github/codeql/pull/15766.

**Review by michaelnebel — DISMISSED** [review]
Looks good to me!

File: java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll [hunk 1]
```diff
@@ -59,12 +59,26 @@ private predicate isInternal(CompilationUnit cu) {
...
+  }
+}
+
 /** Holds if the given callable is not worth modeling. */
 predicate isUninterestingForModels(Callable c) {
   isInTestFile(c.getCompilationUnit().getFile()) or
   isInternal(c.getCompilationUnit()) or
   c instanceof MainMethod or
   c instanceof ToStringMethod or
+  c instanceof LambdaFlowMethod or
```
Comment (atorralba) [inline java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll:81]: Would it make sense to use [`FunctionalInterface`](https://github.com/github/codeql/blob/a97510a2fc3659567dfe7b280d3e782fb3bb4b67/java/ql/lib/semmle/code/java/Type.qll#L1033) instead?

Comment (atorralba) [reply java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll:81]: Hm, nvm, some of those aren't functional interfaces.

Comment (owen-mc) [reply java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll:81]: Hmm, I might be able to use that to replace some of the hard-coded method names. I'll have a play around with it.

Comment (owen-mc) [reply java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll:81]: It gave almost 7000 results! I think it's best to just hard-code a few for now and if we find we want to do more in future then we can come up with a clever way to characterise them.

Comment (atorralba) [reply java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll:82]: Makes sense 👍

**Review by atorralba — APPROVED** [review]

**Review by michaelnebel — APPROVED** [review]
LGTM! 👍

**Pull request merged**